File: tests/src/com/yhsif/onepwd/HmacMd5Test.java
package com.yhsif.onepwd;

import static com.google.common.truth.Truth.assertThat;

import org.junit.Test;

import java.security.NoSuchAlgorithmException;

public class HmacMd5Test {

  @Test
  public void testMd5() throws NoSuchAlgorithmException {
    String str = "asdf";
    byte[] expected = {
      (byte) 0x91, (byte) 0x2e, (byte) 0xc8, (byte) 0x03,
      (byte) 0xb2, (byte) 0xce, (byte) 0x49, (byte) 0xe4,
      (byte) 0xa5, (byte) 0x41, (byte) 0x06, (byte) 0x8d,
      (byte) 0x49, (byte) 0x5a, (byte) 0xb5, (byte) 0x70};
    byte[] md5 = HmacMd5.md5(str.getBytes());
    assertThat(md5).isEqualTo(expected);
  }

  @Test
  public void testHmac() throws NoSuchAlgorithmException {
    String key1 = "asdf";
    String key2 =
        "1234567890123456789012345678901234567890123456789012345678901234567890";
    String message = "google";
    byte[] expected1 = {
        (byte) 0x52, (byte) 0x24, (byte) 0xa1, (byte) 0xb5,
        (byte) 0x60, (byte) 0xb1, (byte) 0xef, (byte) 0x5d,
        (byte) 0xcf, (byte) 0x60, (byte) 0x36, (byte) 0x98,
        (byte) 0xb1, (byte) 0x60, (byte) 0xb5, (byte) 0x92};
    byte[] expected2 = {
        (byte) 0x83, (byte) 0x9d, (byte) 0xce, (byte) 0xbd,
        (byte) 0x94, (byte) 0xac, (byte) 0x5c, (byte) 0x8c,
        (byte) 0x12, (byte) 0xb3, (byte) 0x2a, (byte) 0xfe,
        (byte) 0x33, (byte) 0x5e, (byte) 0x9c, (byte) 0x68};
    byte[] hmac = HmacMd5.hmac(key1, message);
    assertThat(hmac).isEqualTo(expected1);
    hmac = HmacMd5.hmac(key2, message);
    assertThat(hmac).isEqualTo(expected2);
  }

}

File: tests/com/yhsif/onepwd/HmacMd5Test.kt
package com.yhsif.onepwd

import com.google.common.truth.Truth.assertThat

import org.junit.Test

class HmacMd5Test {
  @Test
  fun testMd5() {
    val str = "asdf".toByteArray()
    val expected = byteArrayOf(
      0x91.toByte(), 0x2e.toByte(), 0xc8.toByte(), 0x03.toByte(),
      0xb2.toByte(), 0xce.toByte(), 0x49.toByte(), 0xe4.toByte(),
      0xa5.toByte(), 0x41.toByte(), 0x06.toByte(), 0x8d.toByte(),
      0x49.toByte(), 0x5a.toByte(), 0xb5.toByte(), 0x70.toByte())
    val md5 = HmacMd5.md5(str)
    assertThat(md5).isEqualTo(expected);
  }

  @Test
  fun testHmac() {
    val key1 = "asdf"
    val key2 =
        "1234567890123456789012345678901234567890123456789012345678901234567890"
    val message = "google"
    val expected1 = byteArrayOf(
        0x52.toByte(), 0x24.toByte(), 0xa1.toByte(), 0xb5.toByte(),
        0x60.toByte(), 0xb1.toByte(), 0xef.toByte(), 0x5d.toByte(),
        0xcf.toByte(), 0x60.toByte(), 0x36.toByte(), 0x98.toByte(),
        0xb1.toByte(), 0x60.toByte(), 0xb5.toByte(), 0x92.toByte())
    val expected2 = byteArrayOf(
        0x83.toByte(), 0x9d.toByte(), 0xce.toByte(), 0xbd.toByte(),
        0x94.toByte(), 0xac.toByte(), 0x5c.toByte(), 0x8c.toByte(),
        0x12.toByte(), 0xb3.toByte(), 0x2a.toByte(), 0xfe.toByte(),
        0x33.toByte(), 0x5e.toByte(), 0x9c.toByte(), 0x68.toByte())
    val hmac1 = HmacMd5.hmac(key1, message)
    assertThat(hmac1).isEqualTo(expected1)
    val hmac2 = HmacMd5.hmac(key2, message)
    assertThat(hmac2).isEqualTo(expected2);
  }
}
